beacon_node/http_api/src/lib.rs

```diff
@@ -1933,6 +1933,77 @@ pub fn serve<T: BeaconChainTypes>(
             })
         });
 
+    /*
+     * builder/states
+     */
+
+    let builder_states_path = eth_v1
+        .and(warp::path("builder"))
+        .and(warp::path("states"))
+        .and(chain_filter.clone());
+
+    // GET builder/states/{state_id}/expected_withdrawals
+    let get_expected_withdrawals = builder_states_path
+        .clone()
+        .and(warp::path::param::<StateId>())
+        .and(warp::query::<api_types::ExpectedWithdrawalsQuery>())
+        .and(warp::path("expected_withdrawals"))
+        .and(warp::path::end())
+        .and(warp::header::optional::<api_types::Accept>("accept"))
+        .and_then(
+            |chain: Arc<BeaconChain<T>>,
+             state_id: StateId,
+             query: api_types::ExpectedWithdrawalsQuery,
+             accept_header: Option<api_types::Accept>| {
+                blocking_response_task(move || {
+                    let proposal_slot = query.proposal_slot.unwrap_or(Slot::new(1));
+                    let (state, execution_optimistic, finalized) = match state_id.state(&chain) {
```

Comment:
I think we can safely `?` the `state_id.state` as the state not found error is already being handled and returned from the `state_id.state` function here:

https://github.com/sigp/lighthouse/blob/cb472d21df859efd0870fb43956fe0fcd3aea905/beacon_node/http_api/src/state_id.rs#L178-L188